src/components/icon-button/IconButton.styles.ts

```diff
@@ -0,0 +1,329 @@
+import styled, { css } from 'styled-components';
+import { FlattenSimpleInterpolation } from 'styled-components/ts3.6';
+
+import { TIconButtonSize, IconButtonLayer, IconButtonState } from './IconButton.types';
+import {
+    ICON_BUTTON_SIZES,
+    ICON_BUTTON_DIMENSIONS,
+    ICON_BUTTON_PADDING,
+    ICON_BUTTON_SPACING,
+} from './IconButton.constants';
+
+function generateSizingStyles(size: TIconButtonSize): FlattenSimpleInterpolation {
+    return css`
+        &[data-size='${size}'] {
+            --horizontal-padding: ${ICON_BUTTON_PADDING[size]}px;
+            --size: ${ICON_BUTTON_DIMENSIONS[size]}px;
+            --child-spacing: ${ICON_BUTTON_SPACING[size]}px;
+        }
+    `;
+}
+
+function getRBGColour({ colour, opacity }: IconButtonLayer): string {
+    if (opacity && opacity >= 0 && opacity < 1) {
+        return `rgba(var(${colour}), ${opacity})`;
+    }
+
+    return `rgb(var(${colour}))`;
+}
+
+function getLayerColours(layer: IconButtonLayer): string {
+    return `--colour-${layer.name}: ${getRBGColour(layer)};`;
+}
+
+function generateStateStyles(stateStyles: IconButtonState[]): FlattenSimpleInterpolation {
+    return css`
+        ${stateStyles
+            .map(({ selector, layers }: IconButtonState) => {
+                if (selector === 'base') {
+                    return `
+                        ${layers.map(getLayerColours).join('\n')}
+                    `;
+                }
+
+                return `
+                    &:${selector} {
+                        ${layers.map(getLayerColours).join('\n')}
+                    }
+                `;
+            })
+            .join('\n')};
+    `;
+}
+
+const StyledIconButton = styled.button`
+    // set component colours
+    --colour-primary-rgb: var(--button-colour-primary-rgb, 61, 60, 64);
+    --colour-inverted-primary-rgb: var(--button-colour-inverted-primary-rgb, 255, 255, 255);
```

Comment:
are you highly opinionated about using `inverted` here?
@nevyangelova and I have already used `accent` as a "contrast" color to the main colors.

WDYT?

Reply:
Again, this is from terminology defined by Design.

In this particular case, the point is that this button can be used on a light or dark background within the same overall light or dark theme. The differences between the normal and inverse button are enough that it's not an exact inverse. Take a look at the designs: https://www.figma.com/file/07JlWIGlhklLCJzfrtBhVI/Components-Buttons-Copy?node-id=1111%3A767